src/main/java/org/prebid/server/bidder/brightroll/BrightrollBidder.java

```diff
@@ -170,6 +170,12 @@ private static boolean isImpValid(Imp imp, List<BidderError> errors) {
      * Updates {@link Imp} {@link Banner} and/or {@link Video}.
      */
     private Imp updateImp(Imp imp, PublisherOverride publisherOverride) {
+        final Imp.ImpBuilder impBuilder = imp.toBuilder();
```

Comment:
Not critical, but is it possible we don't need to update the imp?
In this case `toBuilder() -> build()` operations is redundant.

Reply:
Imp has to be updated in atleast one of the scenarios. Publishers need almost all of those configuration as there is no exchange in between to configure those. So imp update is a required one.

Reply:
Yeah, but i mean can it be the case when we don't need to create `builder` at all.

For example in prev. implementation we had 2 conditions: 
- `if (banner != null) {...}`
- `if (video != null && publisherOverride != null) {...}`
If those conditions don't pass - we just return existing `imp` without any modifications.

You've added new condition `if (publisherOverride != null && publisherOverride.getBidFloor() != null) {...}` which may not  fire too. So, builder is not required in this case.

Reply:
yes, I added just as a precaution. One or another case need to be executed atleast in this block as I mentioned above, publishers need those configuration since no exchange is present. BusinessInsider need everything. May be some other publisher in future may pass bid floor, but need to go thru banner/video sections. So it is big chance that we need builder to be created all the time.

Reply:
Honestly, yes. If we can avoid creating builders, just let's do it.
Please, consider this approach just as an idea:
```
private Imp updateImp(Imp imp, PublisherOverride publisherOverride) {
    final BigDecimal bidFloor = publisherOverride != null && publisherOverride.getBidFloor() != null
            ? publisherOverride.getBidFloor()
            : null;
    final List<Integer> impBattr = publisherOverride != null && publisherOverride.getImpBattr() != null
            ? publisherOverride.getImpBattr()
            : null;

    final Banner banner = imp.getBanner();
    if (banner != null) {
        final boolean noSizes = banner.getW() == null && banner.getH() == null
                && CollectionUtils.isNotEmpty(banner.getFormat());

        if (bidFloor != null || impBattr != null || noSizes) {
            final Imp.ImpBuilder impBuilder = imp.toBuilder();

            if (bidFloor != null) {
                impBuilder.bidfloor(bidFloor);
            }
            if (impBattr != null || noSizes) {
                impBuilder.banner(updateBanner(banner, impBattr, noSizes));
            }

            return impBuilder.build();
        }
    }
    final Video video = imp.getVideo();
    if (video != null && (bidFloor != null || impBattr != null)) {
        final Imp.ImpBuilder impBuilder = imp.toBuilder();

        if (bidFloor != null) {
            impBuilder.bidfloor(bidFloor);
        }
        if (impBattr != null) {
            impBuilder.video(updateVideo(video, impBattr));
        }

        return impBuilder.build();
    }

    return imp;
}

private static Banner updateBanner(Banner banner, List<Integer> impBattr, boolean noSizes) {
    final Banner.BannerBuilder bannerBuilder = banner.toBuilder();

    if (impBattr != null) {
        bannerBuilder.battr(impBattr);
    }
    if (noSizes) {
        // update banner with size from first format
        final Format firstFormat = banner.getFormat().get(0);
        bannerBuilder
                .w(firstFormat.getW())
                .h(firstFormat.getH());
    }

    return bannerBuilder.build();
}

private static Video updateVideo(Video video, List<Integer> impBattr) {
    return video.toBuilder()
            .battr(impBattr)
            .build();
}
```
Here we'll create `imp.builder` on demand if one of several conditions fired.

Reply:
Ok, sure. Will do. thanks.